tests/components/drop_connect/test_sensor.py

```diff
@@ -26,36 +27,41 @@
     TEST_DATA_SOFTENER,
     TEST_DATA_SOFTENER_RESET,
     TEST_DATA_SOFTENER_TOPIC,
+    config_entry_filter,
+    config_entry_hub,
+    config_entry_leak,
+    config_entry_protection_valve,
+    config_entry_pump_controller,
+    config_entry_ro_filter,
+    config_entry_softener,
 )
 
 from tests.common import async_fire_mqtt_message
 from tests.typing import MqttMockHAClient
 
 
-async def test_sensors_hub(
-    hass: HomeAssistant, config_entry_hub, mqtt_mock: MqttMockHAClient
-) -> None:
+async def test_sensors_hub(hass: HomeAssistant, mqtt_mock: MqttMockHAClient) -> None:
     """Test DROP sensors for hubs."""
-    config_entry_hub.add_to_hass(hass)
+    config_entry_hub().add_to_hass(hass)
     assert await async_setup_component(hass, DOMAIN, {})
     await hass.async_block_till_done()
 
     current_flow_sensor_name = "sensor.hub_drop_1_c0ffee_water_flow_rate"
-    hass.states.async_set(current_flow_sensor_name, STATE_UNKNOWN)
+    assert hass.states.get(current_flow_sensor_name).state == STATE_UNKNOWN
```

Comment:
These assertions that the state is STATE_UNKNOWN prior to the MQTT data being processed are unnecessary and cause the tests to fail.  The previous code forced a consistent unknown state prior to processing the MQTT data and ensured that previous tests could not influence the current test.  I just assumed that was normal behavior for the MockConfigEntry() construct and it was my responsibility to ensure a consistent state at the beginning of each test.  If these assertions are removed (throughout) or reverted to sets then the tests pass.

Reply:
We should not set the state in tests. With https://github.com/ChandlerSystems/dropmqttapi/pull/1 these tests should pass. In the old situation, if 2 devices share an attribute like `battery`, and one value is updated, then all devices with this sensor will get this update, because it is serviced from the cache. This is not okay.
I believe it is fine to assert on the initial state.

Reply:
I just submitted https://github.com/home-assistant/core/pull/106978 which merges your suggested change into the dropmqttapi module.  Thanks for identifying that issue.